lightning/src/chain/mod.rs

```diff
@@ -57,6 +57,81 @@ pub trait Listen {
 	fn block_disconnected(&self, header: &BlockHeader, height: u32);
 }
 
+/// The `Confirm` trait is used to notify when transactions have been confirmed on chain or
+/// unconfirmed during a chain reorganization.
+///
+/// Clients should use this interface instead of [`Listen`] if they receive transaction data only
+/// relevant to that identified by [`Filter`]. For instance, an Electrum client may implement
+/// [`Filter`] by subscribing to activity related to registered transactions and outputs. Upon
+/// notification, it would pass along the relevant transactions using this interface.
+///
+/// The intended use is as follows:
```

Comment:
This is a bit confusing as it encompasses both actions thee implementor of this trait is taking, as well as actions the user of it is taking. Maybe the tense can be rejiggered to better describe who is doing what.